lib/pos_transaction_register_test.go

```diff
@@ -26,21 +27,225 @@ func TestSanityCheckTransactionRegister(t *testing.T) {
+	require.Equal(uint64(0), feeTimeBucket.totalTxnsSizeBytes)
+	require.Equal(true, feeTimeBucket.Empty())
+
+	// FeeTimeBucket with a single transaction
+	txn := &MempoolTx{
+		FeePerKB:    100000,
+		Added:       time.UnixMicro(1000000),
+		Hash:        NewBlockHash(RandomBytes(32)),
+		TxSizeBytes: 100,
+	}
+
+	require.Nil(feeTimeBucket.AddTransaction(txn))
+	require.Equal(1, len(feeTimeBucket.GetTransactions()))
+	require.Equal(txn.TxSizeBytes, feeTimeBucket.totalTxnsSizeBytes)
+	require.Equal(false, feeTimeBucket.Empty())
+
+	// Try adding the same transaction again.
+	require.Nil(feeTimeBucket.AddTransaction(txn))
+	require.Equal(1, len(feeTimeBucket.GetTransactions()))
+	require.Equal(txn.TxSizeBytes, feeTimeBucket.totalTxnsSizeBytes)
+	require.Equal(false, feeTimeBucket.Empty())
+
+	// Remove non-existing transaction from non-empty FeeTimeBucket.
+	txn2 := &MempoolTx{
+		FeePerKB: 10050,
+		Added:    time.UnixMicro(1000050),
+		Hash:     NewBlockHash(RandomBytes(32)),
+	}
+	feeTimeBucket.RemoveTransaction(txn2)
+	require.Equal(1, len(feeTimeBucket.GetTransactions()))
+	require.Equal(txn.TxSizeBytes, feeTimeBucket.totalTxnsSizeBytes)
+	require.Equal(false, feeTimeBucket.Empty())
+	require.Equal(true, feeTimeBucket.Includes(txn))
+	require.Equal(false, feeTimeBucket.Includes(txn2))
+
+	// Remove existing transactions from FeeTimeBucket.
+	feeTimeBucket.RemoveTransaction(txn)
+	require.Equal(0, len(feeTimeBucket.GetTransactions()))
+	require.Equal(uint64(0), feeTimeBucket.totalTxnsSizeBytes)
+	require.Equal(true, feeTimeBucket.Empty())
+	require.Equal(false, feeTimeBucket.Includes(txn))
+	require.Equal(false, feeTimeBucket.Includes(txn2))
+
+	// Remove non-existing transaction from empty FeeTimeBucket.
+	feeTimeBucket.RemoveTransaction(txn2)
+	require.Equal(0, len(feeTimeBucket.GetTransactions()))
+	require.Equal(uint64(0), feeTimeBucket.totalTxnsSizeBytes)
+	require.Equal(true, feeTimeBucket.Empty())
+	require.Equal(false, feeTimeBucket.Includes(txn))
+	require.Equal(false, feeTimeBucket.Includes(txn2))
+}
+
+func TestTransactionRegisterPrune(t *testing.T) {
+	seed := int64(111)
+	testCases := 1000
+	feeRange := uint64(10000)
+	timestampRange := uint64(10000)
+
+	require := require.New(t)
+	rand := rand.New(rand.NewSource(seed))
+	globalParams := _testGetDefaultGlobalParams()
+	txnPool := _testGetRandomMempoolTxns(rand, globalParams.MinimumNetworkFeeNanosPerKB, feeRange, 1000, timestampRange, testCases)
+
+	txnRegister := NewTransactionRegister(globalParams)
+	totalSize := uint64(0)
+	for _, tx := range txnPool {
+		require.Nil(txnRegister.AddTransaction(tx))
+		totalSize += tx.TxSizeBytes
+	}
+
+	// Try pruning 0 bytes
+	txns, err := txnRegister.PruneToSize(txnRegister.totalTxnsSizeBytes)
+	require.Nil(err)
+	require.Equal(0, len(txns))
+
+	// Remove a single transaction
+	txns, err = txnRegister.PruneToSize(txnRegister.totalTxnsSizeBytes - 1)
+	require.Nil(err)
+	require.Equal(1, len(txns))
```

Comment:
You could use `require.Len` instead here if you prefer that syntax - I'm okay w/ either, just make it a little clearer that this is a length check

Reply:
Cool! Didn't know about this one. I'm gonna start using it.